pandas/core/groupby/groupby.py

```diff
@@ -810,7 +812,18 @@ def __iter__(self) -> Iterator[tuple[Hashable, NDFrameT]]:
         for each group
         """
         keys = self.keys
+        level = self.level
         result = self.grouper.get_iterator(self._selected_obj, axis=self.axis)
+        if isinstance(level, list) and len(level) == 1:
+            # GH 51583
+            warnings.warn(
+                "Initializing a Groupby object with a length-1 list "
```

Comment:
This is probably fine, but I wonder if "Initializing" might be confusing since users never call e.g. DataFrameGroupBy(...). I think "Creating" would be better as it's divorced from `__init__`.

Reply:
I agree, I'll fix it in the next commit!